src/main/java/org/teragrep/JavaPcre.java

```diff
@@ -147,8 +216,12 @@ public void pcre2_singlematch_java(String a, int b){
         int ind = 0;
 
 
-        LibJavaPcre.RegexStruct.ByValue regex_val = LibJavaPcre.INSTANCE.pcre2_single_jmatch(subject, re, offset, match_options);
-        if (regex_val.numVals == 0) {
+        LibJavaPcre.RegexStruct.ByValue regex_val = LibJavaPcre.INSTANCE.pcre2_single_jmatch(subject, re, offset, match_options, mcontext);
+        if (regex_val.rc < 0) {
+            switch(regex_val.rc){
+                case -1: LOGGER.warn("No match"); break; // rc = -1 should be equal to rc = PCRE2_ERROR_NOMATCH in C.
```

Comment:
is no-match common scenario? if yes perhaps LOGGER.debug would be enough. Logging from a library on info level, if there is no real need, is a bit problematic.

Reply:
no-match error is triggered every time at the end when trying to find all match groups. Changed to debug.